### PR TITLE
Mark more types and Driver state as internal/@_spi(Testing)

### PR DESCRIPTION
**Pull request by artemcm**

### CODE REVIEW

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -122,101 +122,101 @@ public struct Driver {
...
-  public let inputFiles: [TypedVirtualPath]
+  @_spi(Testing) public let inputFiles: [TypedVirtualPath]
 
   /// The last time each input file was modified, recorded at the start of the build.
-  public let recordedInputModificationDates: [TypedVirtualPath: Date]
+  @_spi(Testing) public let recordedInputModificationDates: [TypedVirtualPath: Date]
 
   /// The mapping from input files to output files for each kind.
-  internal let outputFileMap: OutputFileMap?
+  @_spi(Testing) public let outputFileMap: OutputFileMap?
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Driver/Driver.swift:137]: This increased visibility from internal to public/spi.

**Review by cltnschlosser — COMMENTED** [review]
What's the difference/benefit of using spi over internal and `@testable`?

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 1 above)
Comment (artemcm) [reply Sources/SwiftDriver/Driver/Driver.swift]: 🤦‍♂️ Thank you!

**Comment by artemcm** [comment]
> What's the difference/benefit of using spi over internal and `@testable`?

One of the key benefits is that `@testable` is not compatible with textual interfaces and we would really like to build SwiftDriver with library-evolution quite soon. Still using `@testable` would mean building twice, once for testing and then again for distribution. 

A little bit more reasoning here:
https://forums.swift.org/t/the-future-of-testable/9655/14

**Comment by artemcm** [comment]
@swift-ci please test

**Review by owenv — APPROVED** [review]
Nice!

**Comment by artemcm** [comment]
@swift-ci please test

**Pull request merged**